model:
  model: "qwen3:4b"
  # See: https://python.langchain.com/api_reference/ollama/chat_models/langchain_ollama.chat_models.ChatOllama.html for a complete list of available parameters.
  temperature: 0.1
  max_tokens: 1000
  top_p: 0.95
  top_k: 40

router:
  prompt: |
    You are an assistant who has to choose which expert to route the question to.
    Experts are: 
    {experts}
    Based on the question, choose the expert that is most relevant to answer it. Respond with the name of the expert only.
    /no_think


#memory: True | False # enable or disable memory

incremental: False  # True | False 
                    # If True, the agent is updated keeping the same structure, and modifying the specified settings. If False, the agent is completely reloaded with the new settings.

chains:
  historian:
    type: "rag"
    description: |
      answers questions about history, historical events, periods, figures and culture. It also answer questions about the current geopolitical situation, including current world leaders, religious figures, and political events.
    prompt: |
      You are a helpful assistant. Here is the conversation so far:
      {history}
      Answer all questions to the best of your ability, using the provided context. Context: {context} 
      /no_think
    context:
      type: "text" # only text is supported for now
      content: |
        On the 8th of May 2025, the US cardinal Robert Francis Prevost was elected pope, taking the name of Pope Leo XIV. He is the follower of Pope Francis, deceased on the 21th of April 2024.
    text_splitter:
      chunk_size: 500
      chunk_overlap: 100
    embedding_model:
      model_name: "sentence-transformers/all-mpnet-base-v2"
      device: cpu
    faiss:
      index_dir: faiss_index_historian
      search:
        k: 3
                    
  fallback:
    type: "chat"
    description: |
      answers on general questions, when no expert is found.
    prompt: |
      You are a helpful assistant. Here is the conversation so far: 
      {history}.
      Answer all questions to the best of your ability. 
      /no_think

  weather:
    type: "tool"
    description: |
      provides current weather information for the specified location.
    location: "New York"
    language: "en" # ["en", "it"]

  datetime:
    type: "tool"
    description: |
      provides current date, month, and season information.
    language: "en" # ["en", "it"]